lib/galaxy/jobs/__init__.py

```diff
@@ -1970,28 +1980,28 @@ def get_output_basenames(self):
         return [os.path.basename(str(fname)) for fname in self.get_output_fnames()]
 
     def get_output_fnames(self):
-        if self.output_paths is None:
```

Comment:
`output_paths` is `[]` now instead of `None`

Reply:
This is a caching technique: You set the unset value to `None`, and then you call an expensive method that generates something that is not None and you store the result of that method. Next time you call the method you know you don't need to re-run the method because `self.output_paths` is not None. But there may be cases where `compute_outputs` generates an empty list, in which case we don't want to re-run `compute_paths`, and that breaks here.

Reply:
Got it, thank you! I've changed it back, please take a look